src/Query.php

```diff
@@ -28,10 +29,24 @@
  */
 class Query implements Filterable, LimitOffsetInterface, OrderByInterface, Paginatable, IteratorAggregate
 {
+    use Events;
     use Filters;
     use LimitOffset;
     use OrderBy;
 
+    /**
+     * Event raised after assembling a {@link Select} object for the query
+     *
+     * **Example usage:**
+     *
+     * ```
+     * $query->on(Query::ON_EXECUTE, function (Select $select) {
```

Comment:
This mentions the wrong event, an invalid one even?

Reply:
Yeah. I named it execute before and forgot to rename here 😆